flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -322,29 +325,24 @@ public void run() {
 	 */
 	@Test
 	public void testRequestBackoffConfiguration() throws Exception {
-		ResultPartitionID[] partitionIds = new ResultPartitionID[] {
-			new ResultPartitionID(),
-			new ResultPartitionID(),
-			new ResultPartitionID()
+		IntermediateResultPartitionID[] partitionIds = new IntermediateResultPartitionID[] {
+			new IntermediateResultPartitionID(),
+			new IntermediateResultPartitionID(),
+			new IntermediateResultPartitionID()
 		};
 
-		InputChannelDeploymentDescriptor[] channelDescs = new InputChannelDeploymentDescriptor[]{
+		ResourceID localLocation = new ResourceID("local");
+		ShuffleDeploymentDescriptor[] channelDescs = new ShuffleDeploymentDescriptor[]{
 			// Local
-			new InputChannelDeploymentDescriptor(
-				partitionIds[0],
-				ResultPartitionLocation.createLocal()),
+			createSddWithLocalConnection(partitionIds[0], localLocation, 10000),
 			// Remote
-			new InputChannelDeploymentDescriptor(
-				partitionIds[1],
-				ResultPartitionLocation.createRemote(new ConnectionID(new InetSocketAddress("localhost", 5000), 0))),
+			createSddWithLocalConnection(partitionIds[1], new ResourceID("remote"), 5000),
```

Comment:
What does `...WithLocalConnection` mean? Don't you want to create a remote `ShuffleDeploymentDescriptor`?

Reply:
refactored to `ShuffleDeploymentDescriptorBuilder`